stmhal/mphalport.h

```diff
@@ -6,10 +6,16 @@
 // go in some MCU-specific header, but for now it lives here.
 #if defined(MCU_SERIES_F4)
 #define MP_HAL_UNIQUE_ID_ADDRESS (0x1fff7a10)
+#define MP_HAL_CLEANINVALIDATECACHE(addr, size)
+#define MP_HAL_CLEANCACHE(addr, size)
 #elif defined(MCU_SERIES_F7)
 #define MP_HAL_UNIQUE_ID_ADDRESS (0x1ff0f420)
+#define MP_HAL_CLEANINVALIDATECACHE(addr, size) (SCB_CleanInvalidateDCache_by_Addr((uint32_t*)(((uint32_t)addr ) & ~(0x1F)), (size) + ((uint32_t)addr - (((uint32_t)addr ) & ~(0x1F)))))
```

Comment:
I think it does the same result, but it's perhaps a little more simple.
But the casts are still needed, otherwise, the compiler complains about  implicit casts.